misc/python/materialize/cli/lint-cargo.py

```diff
@@ -0,0 +1,57 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+"""Check our set of Cargo.toml files for issues"""
+
+import sys
+from pathlib import Path
+from pprint import pprint
+from typing import Dict, List, Optional
+
+from materialize.cargo import Crate, Workspace
+
+
+def check_rust_versions() -> bool:
+    """Checks that every crate has a minimum specified rust version, and furthermore,
+    that they are all the same."""
+
+    def is_relevant(crate: Crate) -> bool:
+        # The `play` directory is for
+        # random miscellaneous stuff that isn't part of
+        # the Materialize product proper; as such, we shouldn't
+        # lint it.
+        return not crate.path.parts[-2] == "play"
+
+    rust_version_to_crate_path: Dict[Optional[str], List[Path]] = {}
+    ws = Workspace(Path.cwd())
```

Comment:
This should be `materialize.ROOT` so that it's not CWD-dependent. That makes the output of `pprint` below a bit ugly, but I think the easiest fix is to store crate names instead of paths.